### PR TITLE
kernel/binary_manager: Add missing BP recovery update and version update

### PR DESCRIPTION
**Pull request by jeongarmy**
- Add a flag for BP recovery
- Set updated BP data in g_bp_info when BP data is updated
- Update version after header verification is done

### CODE REVIEW

File: os/kernel/binary_manager/binary_manager_load.c [hunk 1]
```diff
@@ -219,19 +220,22 @@ static int binary_manager_load(int bin_idx)
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
 			if (bin_idx == BM_CMNLIB_IDX) {
 				ret = binary_manager_read_header(BINARY_COMMON, devpath, &common_header_data, true);
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = common_header_data.version;
 			} else
 #endif
 			{
 				ret = binary_manager_read_header(BINARY_USERAPP, devpath, &user_header_data, true);
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = user_header_data.version;
```
Comment (sunghan-chang) [inline os/kernel/binary_manager/binary_manager_load.c:228]: version -> bin_ver

File: os/kernel/binary_manager/binary_manager_load.c [hunk 2]
```diff
@@ -219,19 +220,22 @@ static int binary_manager_load(int bin_idx)
...
 			}
 			if (ret == BINMGR_OK) {
 				printf("[Binary Loading] %s Header Checking Success\n", BIN_NAME(bin_idx));
 			} else {
+				BIN_VER(bin_idx, BIN_USEIDX(bin_idx)) = 0;
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager_load.c:233]: Could you leave the comment why binary version will be set to 0 for invalid case?

File: os/kernel/binary_manager/binary_manager_load.c (hunk 1 above)
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager_load.c:228]: `common_header_data` has a member of `version`, but `user_header_data` has a `bin_ver`.
How about unifying the member variable name?

Comment (jeongarmy) [reply os/kernel/binary_manager/binary_manager_load.c]: I did to distinguish  2 version of kernel and user, `bin_ver` and `kernel_ver` in user binary header.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**